Dockerfile

```diff
@@ -1,3 +1,8 @@
+FROM node:10.14.2 as builder
+
+COPY . /app
+RUN npm install && npm run bundle && npm run build && rm -rf node_modules
```

Comment:
Let's split this into multiple steps, so we can benefit from Docker's layer caching when package.json doesn't change:

```
COPY package.json  package-lock.json ./
RUN npm install
COPY . ./
RUN npm run build
```

Reply:
There is an issue here though: `npm run bundle` needs the Python packages to really work. Right now we don't load any extensions in the default build, so it's a non issue, but this process will not work for people who use the codebase from extensions.

@emtwo @jezdez this is another case for moving the frontend extensions to use a frontend package manager rather than rely on pip.

Reply:
Yeah, good example. I've looked into plugin patterns in Node land a bit and as always there are many answers to that question. Some examples I found interesting (and very much not exhausting the choices):

- https://github.com/nfarina/homebridge/blob/master/lib/plugin.js an example of a homegrown plugin system that looks for node modules with the "homebridge-" name prefix and loads them

- https://github.com/c9/architect a low-level "everything is a plugin" API that allows flexible customization and simple, native JS syntax, using package.json for configuration ([blogpost](https://medium.com/@ohbytheway/architecture-using-architect-8c8bb7d0277a))

- https://hapijs.com/tutorials/plugins another example of a homegrown system, this time not using the package.json API by default, and with the need to [manually](https://github.com/hapijs/hapi-auth-basic/blob/b73ab49bca00a4a5df1d2172d6cc062f7d56a82c/test/index.js#L26) [register the plugin](https://github.com/hapijs/hapi/blob/ae0255550a6c537f46eaee71b49bbcf3520a8bb2/lib/server.js#L363) (by default at least)

Reply:
I would suggest to reach out to the community and ask for people that may have more experience writing system that are extensible via plugins/extensions. I'm worried that I'm missing the elephant in the room to be honest 😬

Reply:
Thank you for your reviews, and I'm sorry I have no good idea about `npm run bundle`!
So, I add installing step of pip and setuptools tentatively in frontend-builder. :pray:

Reply:
Thank you for the pointers, @jezdez. I will open an issue to bootstrap the discussion.

@koooge sorry for the mixed signals here: I wanted to point out the issue, but I didn't expect you to address it in this pull request. 

I think it's not worth to install the extra stuff for something we currently don't use. The only use of `npm run bundle` is in [mozilla/redash](https://github.com/mozilla/redash) and they have their own Dockerfile...

Reply:
If this statement is not necessary, is it okay to remove?

Reply:
@jezdez #3207